Call on Logging Microservice from Printer API

1234momo opened the pull request.

1234momo commented:
Yes

evanugarte commented:
maybe we should call the log api from the frontend like:

```
print logging service
      ^
      |
      |
      |
      |
frontend      -------------> printer RPC ---------> printer
```

instead of

```
                      print logging service
                                    ^
                                    |
                                    |
                                    |
                                    |
                                    |
frontend      -------------> printer RPC ---------> printer
```

let me know what you think. my bad if i keep switching up ideas.

1234momo commented:
I think calling from the frontend seems better because printerRPC would just need to focus on connecting to the printer while the print logging service can focus on adding a print log

evanugarte commented:
@1234momo sounds good. let's just have the 2D printing APIFunctions call both on the logging service and rpc when the user submits a print.

1234momo commented:
Changes made:
- Called print logging service to log a print request from frontend (src/Pages/2DPrinting/2DPrinting.js)
- Added a printing analytics page (src/Pages/PrintingAnalytics/PrintingAnalytics.js, src/Pages/PrintingAnalytics/PrintingAnalytics.css)
- Added a node file to generate fake print requests (util/mocked-data/PrintLogLineChart.js)

1234momo commented on src/Pages/PrintingAnalytics/PrintingAnalytics.js:
```diff
@@ -0,0 +1,194 @@
+  }
+
+  const lineGraphInfo = {
+    labels: ['8:00', '8:30', '9:00', '9:30', '10:00', '10:30', '11:00',
```
Evan and I wanted to make the labels more dynamic so that you can zoom out and in to show different times and dates, but there weren't any options on react-chartjs-2 that allowed that. If we wanted to change the labels so that more times are tracked, then the labels on the graph would be difficult to view. There weren't many options that we were able to do, so we just stuck with graphing from 8 to 21:30 in 30 min intervals.
